src/support_tests.cc

```diff
@@ -767,6 +770,135 @@ bool test_digest(bool print_all) {
   return true;
 }
 
+#define NUM_TEST_FILES_TO_DIGEST 4
+
+/*
+ * test_digest_multiple() - Similar to test_digest()
+ *
+ * Here we do a simple verification whether the hash-digest produced by
+ * digest_message() is the same if you read-in all input files once
+ * v/s if you read-on each file one-at-a-time and make multiple calls
+ * to digest_message(). Using digest_message(), which is a wrapper around
+ * underlying EVP_Digest*() interfaces, is not additive. This test checks
```

Comment:
@rgerganov & @jlmucb -- Please review this header block and my conclusion that "... is not additive", the way the `digest_message()` is coded.

If you both agree with this test-case, I'd like to retain it, as it closes out a doubt that someone else may have / raise in the future.